tests/assets/eks-pod-identity/pod-default.yaml

```diff
@@ -6,18 +6,88 @@ metadata:
     group: {{.Group}}
 spec:
   containers:
-  - image: registry.k8s.io/pause:3.9
-    imagePullPolicy: IfNotPresent
-    name: pause
-  initContainers:
-  - name: app-init
+  - name: app-with-awsapi
     image: public.ecr.aws/aws-cli/aws-cli:latest
     imagePullPolicy: IfNotPresent
+    env:
+      - name: CLUSTER_NAME
+        value: "{{.ClusterName}}"
+      - name: DIMENSION_NAME
+        value: "{{.MetricDimensionName}}"
+      - name: NAMESPACE
+        value: "{{.MetricNamespace}}"
+      - name: METRIC_LATENCY_NAME
+        value: "{{.MetricLatencyName}}"
+      - name: PERIOD
+        value: "{{.MetricPeriod}}"
     command:
       - sh
       - -c
       - |
+        AUTH_TOKEN=$(cat $AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE)
+        MAX_ATTEMPTS=7
+        INITIAL_DELAY=0.2  # 200ms
+
+        DIMENSION_VALUE=$CLUSTER_NAME
+        METRIC_MAX_RETRIES=5
+        METRIC_RETRY_DELAY=1
+
+        # make 7 attempts on credential fetching with exponential retries, and calculate the time taken
+        # push metrics on time taken on credential fetching
+        # to minimize failure from cloudwatch metrics, add retries on put-metric-data
+        start_epoch=$(date +%s%3N)
+        # fetch credentials
+        for i in $(seq 0 $((MAX_ATTEMPTS - 1))); do
+          if curl -S -H "Authorization: $AUTH_TOKEN" http://169.254.170.23/v1/credentials; then
```

Comment:
do we need to verify that we got a successful response? may be status code 2xx ?